design/pub-sub.md

```diff
@@ -33,25 +33,107 @@ service Yorkie {
 }
 ```
 
-And to manage the event delivery target, we are using
-the [PubSub pattern](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern)
-. You can learn more by looking at
-the [sync package](https://github.com/yorkie-team/yorkie/tree/main/yorkie/backend/sync)
-we are implementing.
-
-The order of operation is as follows.
-
-1. The WatchDocuments API creates a `Subscription` instance with the client and
-   document keys.
-2. Subscription instance internally manages the `DocEvent channel`, and the
-   WatchDocuments API uses a select statement to retrieve events passed to the
-   Subscription instance.
-3. The client can deliver the event to the Subscription instances that are
-   subscribed to the same document through
-   the `Publish(publisherID *time.ActorID, topic string, event DocEvent)`
-   method.
-4. In the select statement mentioned earlier, when an event is confirmed, it is
-   sent to the stream server.
+And to manage the event delivery target, we are using the [PubSub pattern](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern). You can learn more by looking at the [sync package](https://github.com/yorkie-team/yorkie/blob/main/server/backend/sync/pubsub.go) we are implementing.
+![pub-sub pattern vs observer pattern](media/pubsub-pattern.png)
+
+The process of the event pub-sub is as follows:
+![pub-sub event flow](media/pubsub.png)
+
+#### 1. Set SubscriptionsMap
+
+The `watchDocuments` API creates a `Subscription` instance and adds it to the `subscriptionsMapByDocKey`. The `Subscription` instance internally manages the `DocEvent channel`, and a `select` statement is used to retrieve events that are passed to the `Subscription` instance.
+
+```go
+// Subscription represents a subscription of a subscriber to documents.
+type Subscription struct {
+	id         string
+	subscriber types.Client
+	closed     bool
+	events     chan DocEvent
+}
+```
+
+#### 2. Publish Event
+
+The client can send a `DocEvent` to the `Subscription` instances that subscribe to the same document through the `Publish` method.
```

Comment:
How about using publisher instead of client?

```suggestion
The publisher can send a `DocEvent` to the `Subscription` instances that subscribe to the same document through the `Publish` method.
```